soci/soci_index_test.go

```diff
@@ -352,3 +353,15 @@ func TestNewIndexFromReader(t *testing.T) {
 		})
 	}
 }
+
+func TestBuildSociLayer(t *testing.T) {
```

Comment:
Since this test tests for one of the error modes in `buildSociLayer`, maybe it may make sense to name it respectively, e.g. `TestBuildSociLayerErrorModes`, or something like that?